test/csuite/incr_backup/main.c

```diff
@@ -0,0 +1,846 @@
+
+#include "test_util.h"
+
+#include <sys/wait.h>
+#include <signal.h>
+
+#define ITERATIONS 10
+#define MAX_NTABLES 100
+
+#define MAX_KEY_SIZE 100
+#define MAX_VALUE_SIZE 10000
+#define MAX_MODIFY_ENTRIES 10
+#define MAX_MODIFY_DIFF 500
+
+#define URI_MAX_LEN 32
+#define URI_FORMAT "table:t%d-%d"
+#define KEY_FORMAT "key-%d-%d"
+
+static int verbose_level = 0;
+static uint64_t seed = 0;
+
+static void usage(void) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+
+/*
+ * Note: set this to true to copy incremental files completely.
+ */
+static bool slow_incremental = false;
+
+/* TODO: rename and drop are not currently working, they give resource busy. */
+static bool do_rename = false;
+static bool do_drop = false;
+
+#define VERBOSE(level, fmt, ...)      \
+    do {                              \
+        if (level <= verbose_level)   \
+            printf(fmt, __VA_ARGS__); \
+    } while (0)
+
+/*
+ * We keep an array of tables, each one may or may not be in use.
+ * "In use" means it has been created, and will be updated from time to time.
+ */
+typedef struct {
+    char *name;            /* non-null entries represent tables in use */
+    uint32_t name_index;   /* bumped when we rename or drop, so we get unique names. */
+    uint64_t change_count; /* number of changes so far to the table */
+    WT_RAND_STATE rand;
+    uint32_t max_value_size;
+} TABLE;
+#define TABLE_VALID(tablep) ((tablep)->name != NULL)
+
+/*
+ * The set of all tables in play, and other information used for this run.
+ */
+typedef struct {
+    TABLE *table;           /* set of potential tables */
+    uint32_t table_count;   /* size of table array */
+    uint32_t tables_in_use; /* count of tables that exist */
+    uint32_t full_backup_number;
+    uint32_t incr_backup_number;
+} TABLE_INFO;
+
+/*
+ * The set of active files in a backup. This is our "memory" of files that are used in each backup,
+ * so we can remove any that are not mentioned in the next backup.
+ */
+typedef struct {
+    char **names;
+    uint32_t count;
+} ACTIVE_FILES;
+
+extern int __wt_optind;
+extern char *__wt_optarg;
+
+/*
+ * The choices of operations we do to each table.
+ */
+typedef enum { INSERT, UPDATE, MODIFY, REMOVE } OPERATION_TYPE;
+
+/*
+ * usage --
+ *     Print usage message and exit.
+ */
+static void
+usage(void)
+{
+    fprintf(stderr, "usage: %s [-h dir] [-S seed] [-v verbose_level]\n", progname);
+    exit(EXIT_FAILURE);
+}
+
+/*
+ * die --
+ *     Called when testutil_assert or testutil_check fails.
+ */
+static void
+die(void)
+{
+    fprintf(stderr,
+      "**** FAILURE\n"
+      "To reproduce, please rerun with: %s -S %" PRIu64 "\n",
+      progname, seed);
+}
+
+/*
+ * key_value --
+ *     Return the key, value and operation type for a given change to a table. The first 10000
+ *     changes to a table are all inserts, the next 10000 are updates of the same records, the next
+ *     10000 are all modifications of the existing records, the next 10000 will be drops. Then we
```

Comment:
I think `drops` should be `removes`.
Overall I really like this concept to know what we expect.